ksqldb-common/src/main/java/io/confluent/ksql/util/Triple.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import java.util.Objects;
+
+public class Triple<T1, T2, T3> {
+
+  public final T1 left;
+  public final T2 middle;
+  public final T3 right;
+
+
+  public static <L, M, R> Triple<L, M, R> of(final L left, final M middle, final R right) {
+    return new Triple<>(left, middle, right);
+  }
+
+  public Triple(final T1 left, final T2 middle, final T3 right) {
+    this.left = left;
+    this.middle = middle;
+    this.right = right;
+  }
+
+  public T1 getLeft() {
```

Comment:
I created classes for each tuple type (`Quadruple`, `Quintuple`, ...), so I was going to add these as cases in `BaseAggregateFunction#determineInputConverter()`. Ten column arguments does seem a bit excessive, though. Maybe stop at five?

Reply:
Five is reasonable. My question was more, were you planning on following the same `getLeft/getRight` pattern?

Reply:
For `Quadruple` and later, I've been using `getFirst()`, `getSecond()`, `getThird()`, and so on since `getLeft()` and similar didn't fit. I used `getLeft()` and `getRight()` for `Triple` since `Pair` uses these terms.